### ISSUE TITLE
Documentation is very unclear. It lacks formulas in many APIs.

### ISSUE THREAD

**Issue by FrankYinXF** [tensorflow/tensorflow#36844]
For example, in SparseCategoricalAccuracy(), the words in this API does not help to give a clear picture to understand what it is doing. Why not give a formula. A formula, associated with an example, is clear enough for this API.

**Comment by ravikyram** [comment]
@FrankYinXF 

Can you please provide the links you are referring to.Thanks!

**Comment by RohanVKashyap** [comment]
Can you please mention which formula you need exactly,I might probably be able to help you with that.

**Comment by ANONYMOUS-GURU** [comment]
@FrankYinXF 
I can relate to you on this ... Let me clarfiy it for you.
The SparseCategoricalEntropy is similar to the CategoricalCrossentropy Function.
The only difference being that the y_true need not be one-hot columns instead they are integer values denoting class number.

**Let's try it with an example:
You have a 3 class prediction example namely cat, dog and monkey.
cat is assigned as your first label        --> 0
dog is assigned as your second label --> 1
monkey as your third label                  --> 2**

Now while predicting through any network the output logits must have a shape of 3 with one for each of the above mentioned class. When you keep the parameter **logits=True** in the SparseCategoricalCrossEntropy function we feed this layer directly in the loss function as y_pred.
Here the y_true which the loss function expects is not a one-hot column of shape (3,) instead a single integer value corresponding to the class i.e. 
0-> cat
1-> dog
2-> monkey

What the above function does is that it first applies a softmax function(if logits=True) and computes class probailities and on these probabilites it applies a crossentropy function. Now the crossentropy here expects inputs of the same shape so sparsecategoricalcrossEntropy makes a one-hot column in a optimized way to compute the loss instead of feeding a one-hot ourself.

It saves us 2 things , first the burden of making a one-hot column every time and also reduces memory usage as we don't need to save labels as one-hot columns. This is especially useful when the number of classes is extremely high say **>1000**.

Now if **logits=False** the function expects softmax probabilities instead of the logits as input.

**Comment by amahendrakar** [comment]
> @FrankYinXF
> 
> Can you please provide the links you are referring to.Thanks!

Any updates regarding this issue? Thanks!

**Comment by amahendrakar** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
